### PR TITLE
Fix #14: Ignore functionality

### LINKED ISSUE #14: Implement file ignoring behavior
Stuff to consider:

1. There must be an `ignore` attribute on `rebar.config`.
2. There _can_ be an `--ignore` command line flag for `rebar3 hank …`
3. There must be a `-hank ignore.` attribute for modules.
4. It would be nice to be able to ignore just _some_ rules with `-hank(…).`
5. We will likely need to differentiate between two versions of ignoring:
  * Not reporting warnings found in the module:
    - If a record field is defined but not used in the module, do not warn.
    - If a record field defined in some `hrl` is used in the module, consider the field *used*.
  * Not considering the module for the analysis:
    - If a record field is defined but not used in the module, do not warn.
    - If a record field defined in some `hrl` is used in the module, do *not* consider the field *used*.

### PR DESCRIPTION
**Pull request by elbrujohalcon**

### CODE REVIEW

File: priv/test_files/ignore/specific_ignore.erl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+-module(no_ignore).
```
Comment (diegomanuel) [inline priv/test_files/ignore/specific_ignore.erl:1]: Just a detail, but it should be renamed to `specific_ignore`, right?

File: test/ignore_SUITE.erl [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        [Warning || Warning = #{file := File} <- Warnings, string:equal(File, "with_ignore.hrl")],
+
+    {comment, ""}.
+
+%% @doc No warning should be emitted for rules ignored with -hank [rule, ...]
+hank_individual_rules(_Config) ->
+    %% Initialize rebar3 state as if we run `rebar3 hank` with the default rules
+    {ok, State} =
+        rebar3_hank:init(
+            rebar_state:new()),
```
Comment (diegomanuel) [inline test/ignore_SUITE.erl:60]: This one is a candidate to be moved to `hank_test_utils`.

**Review by diegomanuel — APPROVED** [review]
LGTM! Just minor comments.

File: priv/test_files/ignore/specific_ignore.erl (hunk 1 above)
Comment (elbrujohalcon) [reply priv/test_files/ignore/specific_ignore.erl:1]: 🤦

**Pull request merged**